File: problems/1217-relative-sort-array/relative-sort-array.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number[]}
 */
const relativeSortArray = (arr1, arr2) => {
    const hashMap = {};
    let remain = [];
    for (let n of arr1) {
        if (arr2.indexOf(n) !== -1) hashMap[n] = (hashMap[n] || 0) + 1;
        else remain.push(n);
    }
    remain.sort((a, b) => a - b);

    let result = [];
    for (let n of arr2) {
        const total = hashMap[n];
        result.push(...Array(total).fill(n));
    }

    return [...result, ...remain];
    
};

File: problems/1217-relative-sort-array/relative-sort-array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        countMap = {}
        remain = []
        for n in arr1:
            if n not in arr2: 
                remain.append(n)
            else:
                if n in countMap:
                    countMap[n] += 1
                else:
                    countMap[n] = 1
        
        remain.sort()
        
        result = []
        for n in arr2:
            result.extend([n] * countMap[n])
        
        result.extend(remain)
        return result


        